src/NCronJob/Registry/RuntimeJobRegistry.cs

```diff
@@ -141,11 +141,44 @@ public interface IRuntimeJobRegistry
 /// <summary>
 /// Represents a recurring job schedule.
 /// </summary>
-/// <param name="JobName">The job name given by the user.</param>
-/// <param name="CronExpression">The cron expression that defines when the job should be executed.</param>
-/// <param name="IsEnabled">Whether the job is enabled or not.</param>
-/// <param name="TimeZone">The timezone that is used to evaluate the cron expression.</param>
-public sealed record RecurringJobSchedule(string? JobName, string CronExpression, bool IsEnabled, TimeZoneInfo TimeZone);
+public sealed record RecurringJobSchedule
+{
+    internal RecurringJobSchedule(JobDefinition jobDefinition)
```

Comment:
I am struggling if I think this is the right approach.

We are given 3 options:
1. A Service is responsible to create RJS (RecurringJobSchedule) from JD (JobDefinition). The way it was before
2. JD has a method `ToRJS`
3. Or the way you took: RJS takes JD as argument.

Using 3. will result in users can't mock the type anymore easily. They would have to resort to reflection or other means.

Therefore I leaning towards 1 or 2.

Reply:
I didn't think about mocking.
I'm going to implement option 2 (which moves the logic back to JobDefinition).

Reply:
Done.

Also made JobDefinition expose an IsDynamic property (and RecurringJobSchedule as well)